### PR TITLE
Feat:  Test coverage for merkle related functions and keccak bug fix 

### LINKED ISSUE #35: dev: add more test coverage for the incremental merkle tree
The actual merkle tree hook contract 
(cf: https://github.com/astraly-labs/hyperlane_starknet/blob/main/contracts/src/contracts/hooks/merkle_tree_hook.cairo) lacks testing for the internal functions. The idea is to try every single function to avoid unexpected outputs.

### PR DESCRIPTION
**Pull request by JordyRo1**
Resolves #35 #37 

### Keccak bug fix

**Implementation of ByteArray for Keccak Function Input: **

We introduced the ByteArray to concatenate the entries array, which serves as the input for our Keccak function.

** Bug Fix for u64 Span Division and Reversal: **

There was a bug causing information loss for the last element of the array during the division into u64 spans and the subsequent reversal. For instance, given the input 0x111111111111111100111223, the build_u64_array function correctly returned [0x1111111111111111, 0x111223]. However, reversing the endianness of this array resulted in [0x1111111111111111, 0x232111], losing the '00' in the process.
This pull request resolves the issue by considering the last two u64 elements of the initial input and ensuring the size of the reversed output matches the size of the last word plus 8 bytes.

###  Test coverage 
Added additional tests for the merkle tree hook, the keccak computation, and fixing existing bugs with the current tests. 

###  TODO
  - [x] Detailed documentation on the computation process for keccak,
  - [x] Code cleanup (remove unused imports, and variables for the modified contracts ONLY)

### CODE REVIEW

**Review by EvolveArt — APPROVED** [review]

File: contracts/src/contracts/libs/message.cairo [hunk 1]
```diff
@@ -51,22 +53,43 @@ pub impl MessageImpl of MessageTrait {
...
+                            ByteData {
+                                value: data.into(), size: u256_word_size(data.into()).into()
+                            }
+                        );
                 },
                 Option::None(_) => { break (); }
             };
         };
+        let mut cur_idx = 0;
+        let cloned = @input;
```
Comment (EvolveArt) [inline contracts/src/contracts/libs/message.cairo:92]: ?

File: contracts/src/tests/hooks/test_merkle_tree_hook.cairo [hunk 2]
```diff
@@ -126,3 +132,49 @@ fn test_count() {
     let count = merkle_tree.count();
     assert_eq!(count, 0);
 }
+
+// // TEST INTERNAL
```
Comment (EvolveArt) [inline contracts/src/tests/hooks/test_merkle_tree_hook.cairo:136]: ?

File: contracts/src/tests/hooks/test_merkle_tree_hook.cairo [hunk 3]
```diff
@@ -126,3 +132,49 @@ fn test_count() {
...
+
+#[test]
+fn test_insert() {
```
Comment (EvolveArt) [inline contracts/src/tests/hooks/test_merkle_tree_hook.cairo:139]: better name

File: contracts/src/utils/keccak256.cairo [hunk 4]
```diff
@@ -437,22 +648,44 @@ mod tests {
...
+            0x61a4bcca63b5e8a4,
+            0x6da3abe2080f75c1,
+            0x6c18467d5838f00b,
+            0x375d9ba4c7c313dd,
+            0x49d35915d0abec0a,
+            0x28990198bb32aa57,
+            0x0e681e7eb41a001c,
+            0x0094c7c36a712671
+        ];
+        let expected_result = array![
```
Comment (EvolveArt) [inline contracts/src/utils/keccak256.cairo:679]: how did we get that?

**Review by EvolveArt — COMMENTED** [review]
why remove the `MerkleLib` ?

File: contracts/src/utils/keccak256.cairo (hunk 4 above)
Comment (JordyRo1) [reply contracts/src/utils/keccak256.cairo:679]: it's just teh reverse endianness of the input span<u64>

**Pull request merged**